hack/arktos_cherrypick.sh

```diff
@@ -0,0 +1,210 @@
+    TOUCH_CMD="touch -mt "
+elif [[ "$OSTYPE" == "linux"* ]]
+then
+    SED_CMD=`which sed`
+    if [ -z $SED_CMD ]
+    then
+        echo "Please install sed"
+        exit 1
+    fi
+    STAT_CMD="stat -c %y "
+    TOUCH_CMD="touch -d "
+else
+    echo "Unsupported OS $OSTYPE"
+    exit 1
+fi
+
+display_usage() {
+    echo "Usage: $0 <optional-arktos-repo-path> <optional-log-directory>"
+    echo "       If optional Arktos repo path is provided, repo setup step will be skipped"
+}
+
+if [ ! -z $2 ]
+then
+    LOGDIR=$2
+    LOGFILE=$LOGDIR/$LOGFILENAME
+fi
+
+if [ ! -z $1 ]
+then
+    if [[ ( $1 == "--help") ||  $1 == "-h" ]]
+    then
+        display_usage
+        exit 0
+    else
+        REPODIRNAME=$1
+        if [ -z $2 ]
+        then
+	    LOGFILE=$REPODIRNAME/../$LOGFILENAME
+        fi
+        rm -f $LOGFILE
+        inContainer=true
+        if [[ -f /proc/1/sched ]]
+        then
+            PROC1=`cat /proc/1/sched | head -n 1`
+            if [[ $PROC1 == systemd* ]]
+            then
+                inContainer=false
+            fi
+        else
+            if [[ "$OSTYPE" == "darwin"* ]]
+            then
+                inContainer=false
+            fi
+        fi
+        if [ "$inContainer" = true ]
+        then
+            echo "WARN: Skipping cherrypick check for in-container build as git repo is not available"
+            echo "WARN: Skipping cherrypick check for in-container build as git repo is not available" >> $LOGFILE
+            exit 0
+        else
+            echo "Running cherrypick check for repo: $REPODIRNAME, logging to $LOGFILE"
+        fi
+    fi
+fi
+
+clone_repo() {
+    local REPO=$1
+    local DESTDIR=$2
+    git clone $REPO $DESTDIR
+}
+
+setup_repos() {
+    if [ -d $TMPDIR ]; then
+        rm -rf $TMPDIR
+    fi
+    mkdir -p $TMPDIR
+    clone_repo $ARKTOS_REPO $REPODIRNAME
+}
+
+get_added_files_list() {
+    pushd $REPODIRNAME
+    local HEAD_COMMIT=`git rev-list HEAD | head -n 1`
+    local MERGED_COMMIT=$( git log --show-signature --oneline | grep "gpg: Signature made" | head -n 1 | cut -c1-7 )
+    echo "MERGED_COMMIT: $MERGED_COMMIT, HEAD_COMMIT: $HEAD_COMMIT"
+    git diff --name-only --diff-filter=A $MERGED_COMMIT $HEAD_COMMIT | \
+        egrep -v "\.git|\.md|\.bazelrc|\.json|\.pb|\.yaml|BUILD|boilerplate|vendor\/" | \
+        egrep -v "perf-tests\/clusterloader2\/" | \
+        egrep -v "staging\/src\/k8s.io\/component-base\/metrics\/" | \
+        egrep -v "staging\/src\/k8s.io\/component-base\/version" | \
+        egrep -v "\.mod|\.sum|\.png|\.PNG|OWNERS|arktos_copyright" > $LOGDIR/added_files
+    git diff --cached --name-only --diff-filter=A | \
+        egrep -v "\.git|\.md|\.bazelrc|\.json|\.pb|\.yaml|BUILD|boilerplate|vendor\/" | \
+        egrep -v "\.mod|\.sum|\.png|\.PNG|OWNERS|arktos_copyright" >> $LOGDIR/added_files || true
+    #grep -F -x -v -f $REPODIRNAME/hack/arktos_copyright_copied_k8s_files $LOGDIR/added_files_git > $LOGDIR/added_files_less_copied
```

Comment:
we can remove these commented lines if they are not needed

Reply:
removed these comments.